src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,455 @@
+    });
+    function convertToAsyncFunction(changes: textChanges.ChangeTracker, sourceFile: SourceFile, position: number, checker: TypeChecker, context: CodeFixContextBase): void {
+        // get the function declaration - returns a promise
+        const functionToConvert: FunctionLikeDeclaration = getContainingFunction(getTokenAtPosition(sourceFile, position)) as FunctionLikeDeclaration;
+        if (!functionToConvert) {
+            return;
+        }
+
+        // add the async keyword
+        changes.insertModifierBefore(sourceFile, SyntaxKind.AsyncKeyword, functionToConvert);
+
+        const synthNamesMap: Map<[Identifier, number]> = createMap(); // number indicates the number of times it is used after declaration
+        const lastDotThenMap: Map<boolean> = createMap();
+
+        const functionToConvertRenamed: FunctionLikeDeclaration = renameCollidingVarNames(functionToConvert, checker, synthNamesMap, context);
+        findLastDotThens(functionToConvertRenamed, lastDotThenMap, checker);
+
+        const returnStatements = getReturnStatementsWithPromiseCallbacks(functionToConvertRenamed);
+        let allNewNodes: Map<Node[]> = createMap();
+        for (const statement of returnStatements) {
+            if (isCallExpression(statement)) {
+                const newNodes = parseCallback(statement, checker, statement, synthNamesMap, lastDotThenMap, context);
+                if (newNodes.length) {
+                    allNewNodes = allNewNodes.set(String(getNodeId(statement)), newNodes);
+                }
+            }
+            else {
+                forEachChild(statement, function visit(node: Node) {
+                    if (isCallExpression(node)) {
+                        const newNodes = parseCallback(node, checker, node, synthNamesMap, lastDotThenMap, context);
+                        if (newNodes.length) {
+                            allNewNodes = allNewNodes.set(String(getNodeId(statement)), newNodes);
+                        }
+                    }
+                    else if (!isFunctionLike(node)) {
+                        forEachChild(node, visit);
+                    }
+                });
+            }
+        }
+
+        replaceNodes(changes, sourceFile, returnStatements, allNewNodes);
+    }
+
+    function replaceNodes(changes: textChanges.ChangeTracker, sourceFile: SourceFile, oldNodes: Node[], allNewNodes: Map<Node[]>) {
+        allNewNodes.forEach((value, key) => {
+            for (const statement of oldNodes) {
+                if (String(getNodeId(statement)) === key) {
+                    changes.replaceNodeWithNodes(sourceFile, statement, value);
+                    break;
+                }
+            }
+        });
+    }
+
+    function findLastDotThens(func: FunctionLikeDeclaration, lastDotThen: Map<boolean>, checker: TypeChecker) {

```

Comment:
Comments, particularly around what "true" and "false" mean in the output of the map.

It looks like this is only called once - can it just return the map instead of taking a parameter to mutate?